### PR TITLE
Added feature to crop data for emgacfdr

### PR DESCRIPTION
**Pull request by ninahakansson**
Added possibility to crop data between start_line and end_line.

Left to do:

Update or do not read midnight_line, overlap_free_end and overlap_free_start  as these are no longer correct after cropping data.
Add tests.

### CODE REVIEW

File: level1c4pps/eumgacfdr2pps_lib.py [hunk 1]
```diff
@@ -195,7 +196,31 @@ def set_header_and_band_attrs(scene):
...
+    """Crop datasets and update start_time end_time objects."""
+    start_time_dt64 = scene[time_key].values[start_line]
+    end_time_dt64 = scene[time_key].values[end_line-1]
+    start_time = dt64_to_datetime(start_time_dt64)
+    end_time = dt64_to_datetime(end_time_dt64)
+    print("time to crop the data")
+    for ds in scene.keys():
+        print(ds)
+        if 'y' in scene[ds].dims:
+            scene[ds] = scene[ds].isel(y=slice(start_line, end_line))
```
Comment (sfinkens) [inline level1c4pps/eumgacfdr2pps_lib.py]: I think it needs to be `end_line + 1` for the end line to be included. My fault initially, sorry.

Comment (sfinkens) [reply level1c4pps/eumgacfdr2pps_lib.py]: And the corresponding end timestamp would be `end_time_dt64 = scene[time_key].values[end_line]`

File: level1c4pps/eumgacfdr2pps_lib.py [hunk 2]
```diff
@@ -229,6 +256,12 @@ def process_one_file(eumgacfdr_file, out_path='.', reader_kwargs=None, engine='h
     # Handle gac specific datasets qual_flags and scanline_timestamps
     update_ancilliary_datasets(scn_)
 
+    if start_line is not None and end_line is not None:
+        # Crop after all renaming of variables are done
+        # Problems to rename if cropping is done first.
```
Comment (sfinkens) [inline level1c4pps/eumgacfdr2pps_lib.py]: Which problems exactly did you encounter?

Comment (ninahakansson) [reply level1c4pps/eumgacfdr2pps_lib.py]: This (in function rename_latitude_longitude in file __init__.py) no longer works:
 scene['lat'] = scene[lat_name_satpy]

Comment (sfinkens) [reply level1c4pps/eumgacfdr2pps_lib.py]: Hmm, that's strange.

**Comment by ninahakansson** [comment]
I added "removal" of broken data. The data stays but values are set to nodata. @sfinkens  do you have any suggestion on how to do this in a better more xarray-ish way?

**Pull request merged**